### ISSUE TITLE
Include a Check for Bad Product IDs?

### ISSUE THREAD

**Issue by reaktorsicherheit** [UMSKT/UMSKT#81]
I wonder whether it would be useful to include a check for Bad Product IDs.

These are IDs that would be rejected as invalid during a Service Pack installation:

>As of 2005/10/22, Microsoft states that the following are invalid PIDs:
>
>    XXXXX-640-0000356-23XXX
>    XXXXX-640-2001765-23XXX
>
>The above two product IDs are known as "Devil's Own", and usually have a product key starting with "FCKGW".
>
>    XXXXX-640-643718X-23XXX
>    XXXXX-641-309376X-23XXX
>    XXXXX-642-064580X-23XXX
>    XXXXX-642-464364X-23XXX
>    XXXXX-643-334701X-23XXX
>    XXXXX-644-081772X-23XXX
>    XXXXX-644-451265X-23XXX
>    XXXXX-644-874896X-23XXX
>    XXXXX-644-933704X-23XXX
>    XXXXX-644-962396X-23XXX
>    XXXXX-645-833254X-23XXX
>    XXXXX-645-994962X-23XXX
>    XXXXX-646-031843X-23XXX
>    XXXXX-646-104081X-23XXX
>    XXXXX-646-105103X-23XXX
>    XXXXX-647-318838X-23XXX
>    XXXXX-647-592029X-23XXX
>    XXXXX-647-677834X-23XXX
>    XXXXX-648-301691X-23XXX
>    XXXXX-648-819992X-23XXX
>    XXXXX-649-106765X-23XXX
>    XXXXX-649-941392X-23XXX
>    XXXXX-650-292312X-23XXX
>
>These PIDs are detected when you attempt to install SP1, 2 or 3.

https://wiki.lunarsoft.net/wiki/Product_IDs

**Comment by thepwrtank18** [comment]
tl;dr: you don't wanna go down that route

Microsoft has never made a public list of product ID's. To see if one is valid, we need to see if that product ID works or not in all the products Microsoft has, and then make a table specifying what product has what ID's.

Considering there are  255 BINK ID's * 1000 Channel ID's = 255,000 combinations, that's gonna take a while. Not to say that you can't filter that out (if you know the BINK ID you've already reduced it to 1,000), but it's still time consuming.

**Comment by Neo-Desktop** [comment]
the most we can do is include the ranges seen in pidgen.dll - it's not entirely encompassing by any means, but I suppose it's something. this is something that will be supported in the refactor
